### ISSUE TITLE
Imports behind TYPE_CHECKING does not trigger undefinedVariable

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#2440]
**Describe the bug**
Consider followind code:

```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import os

print(os.path.curdir)
```

At this point os is undefined, but pyright thinks it is, becouse it was imported for TYPE_CHECKING.

**Comment by erictraut** [comment]
This is correct and intended behavior. The `TYPE_CHECKING` symbol is meant as a way to "lie" to the type checker. It should therefore be used with great care because it can lead to different behaviors at runtime versus type checking time. This is such an example.
